packages/agoric-cli/lib/start.js

```diff
@@ -111,6 +204,283 @@ export default async function startMain(progname, rawArgs, powers, opts) {
     return ps;
   }
 
+  async function startLocalChain(profileName, startArgs, popts) {
+    const IMAGE = `agoric/agoric-sdk`;
+
+    const portNum = startArgs[0] === undefined ? CHAIN_PORT : startArgs[0];
+    if (`${portNum}` !== `${Number(portNum)}`) {
+      log.error(`Argument to local-chain must be a port number`);
+      return 1;
+    }
+    agServer += `-${portNum}`;
+
+    if (popts.pull) {
+      const status = await pspawn('docker', ['pull', IMAGE]);
+      if (status) {
+        return status;
+      }
+    }
+
+    if (popts.reset) {
+      log(chalk.green(`removing ${agServer}`));
+      // rm is available on all the unix-likes, so use it for speed.
+      await pspawn('rm', ['-rf', agServer]);
+    }
+
+    let chainSpawn;
+    if (popts.sdk) {
+      chainSpawn = (args, spawnOpts = undefined) =>
+        pspawn('ag-chain-cosmos', [`--home=${agServer}`, ...args], spawnOpts);
+    } else {
+      chainSpawn = (args, spawnOpts = undefined, dockerArgs = []) =>
+        pspawn(
+          'docker',
+          [
+            'run',
+            `--volume=${process.cwd()}:/usr/src/dapp`,
+            `--rm`,
+            ...dockerArgs,
+            `-it`,
+            IMAGE,
+            `--home=/usr/src/dapp/${agServer}`,
+            ...args,
+          ],
+          spawnOpts,
+        );
+    }
+
+    if (!(await exists(agServer))) {
+      const status = await chainSpawn([
+        'init',
+        'local-chain',
+        `--chain-id=${CHAIN_ID}`,
+      ]);
+      if (status) {
+        return status;
+      }
+    }
+
+    // Get or create the essential addresses.
+    const addrs = {};
+    for (const keyName of ['delegate0', 'provision']) {
+      /* eslint-disable no-await-in-loop */
+      let capret = showKey(keyName);
+      if (await capret[0]) {
+        const status = await keysSpawn([
+          'keys',
+          'add',
+          keyName,
+          '--keyring-backend=test',
+        ]);
+        if (status) {
```

Comment:
does a truthy status mean there was an error? maybe name it `error` or `failure`?

Reply:
It's the shell "exit status".  Knowing that what better name should I use?

Reply:
ah. hm, maybe `exitStatus` then. "`rc`" could work but it's a bit short.